tests/manage/rgw/test_rgw_routes.py

```diff
@@ -0,0 +1,113 @@
+import logging
+
+import boto3
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import (
+    bugzilla,
+    on_prem_platform_required,
+    skipif_external_mode,
+    tier1,
+)
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.bucket_utils import (
+    s3_delete_object,
+    s3_get_object,
+    s3_put_object,
+)
+from ocs_ci.ocs.exceptions import CommandFailed, UnavailableResourceException
+from ocs_ci.ocs.resources.objectbucket import OBC
+
+log = logging.getLogger(__name__)
+
+
+@on_prem_platform_required
+class TestRGWRoutes:
+    """
+    Test the RGW routes in an ODF cluster
+
+    """
+
+    @skipif_external_mode
+    @bugzilla("2139037")
+    @pytest.mark.polarion_id("OCS-5168")
+    @tier1
+    def test_rgw_routes(self, rgw_bucket_factory):
+        """
+        Test the availability of RGW routes in an ODF cluster
+
+        1. Assert that RGW's service is exposed by both http and https routes
+        2. Assert that both endpoints use a TLS termination policy
+        3. Test basic I.O functionality using both endpoints
+
+        """
+        route_obj = ocp.OCP(
+            kind="Route", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+
+        log.info("Getting RGW's HTTP and HTTPS routes")
+        try:
+            http_route = route_obj.get(resource_name=constants.RGW_ROUTE_INTERNAL_MODE)
+        except CommandFailed as ex:
+            log.error("Failed to get RGW's HTTP route!")
+            raise UnavailableResourceException(ex)
+
+        try:
+            https_route = route_obj.get(
+                resource_name=constants.RGW_ROUTE_INTERNAL_MODE_SECURE
+            )
+        except CommandFailed as ex:
+            log.error("Failed to get RGW's HTTPS route!")
+            raise UnavailableResourceException(ex)
+
+        log.info(
+            "Asserting that RGW's service is exposed by both http and https routes"
+        )
+        assert http_route["spec"]["to"]["name"] == constants.RGW_SERVICE_INTERNAL_MODE
+        assert https_route["spec"]["to"]["name"] == constants.RGW_SERVICE_INTERNAL_MODE
+        assert http_route["spec"]["port"]["targetPort"] == "http"
+        assert https_route["spec"]["port"]["targetPort"] == "https"
+
+        log.info("Asserting that both endpoints use a TLS termination policy")
+        assert http_route["spec"]["tls"]["insecureEdgeTerminationPolicy"] is not None
+        assert https_route["spec"]["tls"]["insecureEdgeTerminationPolicy"] is not None
+
+        log.info("Testing basic I.O functionality using both endpoints")
+        rgw_bucket_name = rgw_bucket_factory(amount=1, interface="RGW-OC")[0].name
+        rgw_obc = OBC(rgw_bucket_name)
+
+        for route in http_route, https_route:
+            # Apply the current route to the OBC s3_client
+            route_name = route["metadata"]["name"]
```

Comment:
can we take this opportunity to improve 
```python
class RGW(object):
``` 
and implement this piece of code using RGW class objects hence enabling the re usability.

Reply:
Just to clarify, do you wish to capture the three basic S3 operations that I used here in a method with a name like `test_basic_io` in the RGW class?

Reply:
No I mean.. if you see the current RGW() class its very basic.. we dont initialize class with s3 client or resources. and you do it here in the test code, can you move that part to RGW class init ? basically something similar to what happens in MCG()

Reply:
I considered adding endpoint properties and "get_s3_resource" methods to the RGW class, but since the merge of this test is time sensitive and implementation depends on comparing the usage with an exernal cluster, I think its better to implement in a separate enhancement PR.

We can track and discuss further enhancements to the RGW class in this issue: https://github.com/red-hat-storage/ocs-ci/issues/8601